packages/client/hmi-client/src/components/widgets/tera-related-documents.vue

```diff
@@ -183,62 +185,67 @@ const sendForEnrichment = async () => {
 
 	isLoading.value = false;
 	emit('enriched');
-	await getRelatedDocuments();
+	getRelatedDocuments();
 };
 
 const sendForExtractions = async () => {
 	const selectedResourceId = selectedResources.value?.id ?? null;
-	if (props.assetType === AssetType.Dataset) {
-		isLoading.value = true;
+	isLoading.value = true;
 
+	// Dataset extraction
+	if (props.assetType === AssetType.Dataset) {
 		await extractPDF(selectedResourceId);
-
-		if (!props.assetId) return;
 		await createProvenance({
 			relation_type: RelationshipType.EXTRACTED_FROM,
-			left: props.assetId,
+			left: props.assetId!,
 			left_type: mapAssetTypeToProvenanceType(props.assetType),
 			right: selectedResourceId,
 			right_type: ProvenanceType.Document
 		});
-	} else if (props.assetType === AssetType.Model && selectedResourceId) {
-		await extractVariables(selectedResourceId, [props.assetId]);
 
-		const linkedAmr = await alignModel(props.assetId, selectedResourceId);
-		if (!linkedAmr) return;
+		logger.info('Provenance created after extraction', { showToast: false });
+		emit('extracted');
+	}
+
+	// Model extraction
+	if (props.assetType === AssetType.Model && selectedResourceId) {
+		await extractVariables(selectedResourceId, [props.assetId]);
+		const isAligned = await alignModel(props.assetId, selectedResourceId);
+		if (isAligned) {
+			logger.success('Model aligned after variable extraction.');
+			emit('enriched');
+		} else {
+			logger.warn('Model was not aligned after variable extraction. Please try again.');
+		}
 	}
 
 	isLoading.value = false;
-	emit('extracted');
-	await getRelatedDocuments();
+	getRelatedDocuments();
 };
 
+function getRelatedDocuments() {
+	if (!props.assetType) return;
+
+	const provenanceType = mapAssetTypeToProvenanceType(props.assetType);
+	if (!provenanceType) return;
+
+	getRelatedArtifacts(props.assetId, provenanceType, [ProvenanceType.Document]).then((nodes) => {
```

Comment:
`getRelatedArtifacts` is still an async call - will need to `await` on this (and `getRelatedDocuments`).